scripts/zones/Inner_Horutoto_Ruins/mobs/Goblin_Thug.lua

```diff
@@ -2,10 +2,32 @@
 -- Area: Inner Horutoto Ruins
 --  Mob: Goblin Thug
 -----------------------------------
+local ID = zones[xi.zone.INNER_HORUTOTO_RUINS]
+-----------------------------------
 local entity = {}
+local nmPopChance = 50
 
 entity.onMobDeath = function(mob, player, optParams)
     xi.regime.checkRegime(player, mob, 647, 2, xi.regime.type.GROUNDS)
 end
 
+entity.onMobDespawn = function(mob)
+    local totd = VanadielTOTD()
```

Comment:
treat them as if an NM, instead of reinventing the wheel.

```lua
xi.mob.phOnDespawn(mob, ID.mob.MAGICKED_BONES_PH_DAGGER, 50, 1) -- 50% chance, pure lottery (meaning no cooldown)
```

Reply:
I was originally doing this, but it was causing problems. They were spawning on their own with the placeholders being alive still despite spawnType being set to 33. It also was causing an error when trying to call UpdateNMSpawnPoint

Reply:
you did something incorrectly for that to happen. 33 isn't a correct value either.

UpdateNMSpawnPoint error was informing you they have no positions recorded in that table.

the lotto system does also allow spawning immediately by the way.

Reply:
Yeah, I did see that it could spawn immediately normally as well, I just wasn't sure if they were supposed to be immediate or delayed. As for the value of 33, 33 should be 32 (lotto spawn) && 1 (night spawn), which is also the spawntype used for Citipati in the database, unless that is also a bug?

Reply:
I'm pretty sure they are supposed to just replacing the goblin at next pop. that's how all shared spawns work all over the game world (there's several different orcs that share the spawn with Ashmaker GotBlutt's actual placeholder in Yuhgott Grotto, only 1 can be up at a time - not that we do not want to treat every case of this like an NM its just your bones here are special coz they have a drop).

Reply:
I'm gonna narrow this down for you:
> They were spawning on their own with the placeholders being alive
this specifically is user error. no matter what else. nothing else mattered if they were still spawning on their own. Solve that first. You likely gave them other incorrect values besides spawntype, its not the only element that makes them spawn/not spawn.

Reply:
This is functionally what they are doing currently, when the goblin rolls the pop for the bones, it disables its own respawn which then gets re-enabled by the appropriate bones once the bones despawn either naturally or from being killed. This specific scenario is also the only situation I could find where a night only lotto had a placeholder that could also spawn during the day, which seemed to be causing weird behavior.

Reply:
also while spawn_type seems like a flag, afaik everywhere in the code treats it like a value (i.e. you can't use two types of spanw type). Just use "scripted spawn" type and the xi.mob.phOnDespawn teo shared with a 1s NM repop window `xi.mob.phOnDespawn(mob, ID.mob.MAGICKED_BONES_PH_DAGGER, 50, 1, true)` to pop immediately after ph despawn (though i doubt that's how it's supposed to work)

Reply:
> I'm gonna narrow this down for you:
> 
> > They were spawning on their own with the placeholders being alive
> 
> this specifically is user error. no matter what else. nothing else mattered if they were still spawning on their own. Solve that first. You likely gave them other incorrect values besides spawntype, its not the only element that makes them spawn/not spawn.

I tried a bunch of things to try to figure out what was causing it and I couldn't find it for the life of me, and couldn't find an answer for what else would cause them to spawn. spawnType was the only thing I could find or get info about, but if what MowFord says is true as well, the 33 value for Citipati is also wrong then, and that I had been given incorrect information about being able to AND them together to get multiple flags effects. I also was unaware of the phOnDespawn working right with it set to 128 for the scripted spawn, but in retrospect I am not sure why I thought that.

As for the other elements that can cause things to spawn though, could I get some extra info about that? Like I said, I hadn't been able to find anything beyond spawnType really handling that at all.

Reply:
@Thorinori spawntype_night gives sets mobs to spawnable when the zone changes to night

these are PH mobs that only spawn via a script, they need to explicitly only have spawntype_scripted

Reply:
> @Thorinori spawntype_night gives sets mobs to spawnable when the zone changes to night
> 
> these are PH mobs that only spawn via a script, they need to explicitly only have spawntype_scripted

Currently they do only have 128 set, when I was initially trying with 33 they were misbehaving like I said. It could be user error apparently, but like I said I couldn't find anywhere else that would affect how they spawned, especially since it was inconsistent if they misbehaved or not. Like they would behave for a few pops, then suddenly just be there chilling with the PHs.

Reply:
that's part of why using phMobDespawn is so nice. It handles disabling respawn of the PH and enabling respawn of the NM, then flipping them when the NM dies